src/main/kotlin/br/com/creditas/credit_analysis/controller/ClientController.kt

```diff
@@ -123,8 +124,107 @@ class ClientController(
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @Transactional
     fun deleteClientByCpf(@RequestParam cpf: String) {
-        clientRepository.findByCpf(cpf) ?.let {
+        clientRepository.findByCpf(cpf)?. let {
             clientRepository.deleteByCpf(cpf)
         } ?: throw NotFoundException(notFoundMessage)
     }
+
+    @PutMapping("/{id}")
+    @Transactional
+    fun updateClientById(
+        @PathVariable id: UUID,
+        @RequestBody @Valid clientRequest: ClientUpdateRequest
+    ): ClientPF {
+
+        clientRepository.findByid(id)?.let {
+
+            var client = it
+
+            var clientEntity = ClientPF(
+                id = client.id,
+                cpf = clientRequest.cpf,
+                name = clientRequest.name,
+                lastName = clientRequest.lastName,
+                birthDate = clientRequest.birthDate
+            )
+
+            client.address?.let { address -> addressRepository.delete(address) }
```

Comment:
Why not just overwrite the address?

Reply:
Because the relationship between the client and address entities is one-to-one, if I try to change the address without excluding the previous one, it gives an error, because two addresses will be linked to the same client-id.